packages/autocomplete-core/src/createAutocomplete.ts

```diff
@@ -53,6 +56,33 @@ export function createAutocomplete<
       navigator: props.navigator,
       ...setters,
     });
+
+    if (
+      !isAlgoliaInsightsPluginEnabled &&
+      state.context?.algoliaInsightsPlugin?.__automaticInsights &&
+      props.insights !== false
+    ) {
+      const plugin = createAlgoliaInsightsPlugin({
+        __autocomplete_clickAnalytics: false,
+      });
+
+      props.plugins.push(plugin);
+
+      plugin.subscribe?.({
+        ...setters,
+        navigator: props.navigator,
+        refresh,
+        onSelect(fn) {
+          subscribers.push({ onSelect: fn });
+        },
+        onActive(fn) {
+          subscribers.push({ onActive: fn });
+        },
+        onResolve(fn) {
+          subscribers.push({ onResolve: fn });
+        },
+      });
+    }
```

Comment:
Maybe this could be extracted as a reusable function, ie `subscribePlugin(...plugins)`.